doc/read.md

```diff
@@ -0,0 +1,101 @@
+
+The server MUST reply to a successful `READ` command using a `READ` server command, or using an error message.
+
+#### `READ` server command
+
+When sent from a client, the `READ` command signals to the client that the last message read by the user, to the best knowledge of the server, has the specified timestamp. In that case, the command has the following syntax:
+
+    <prefix> READ <target> <timestamp>
+
+The `prefix` is the current server prefix.
+
+#### Command flows
+
+The server SHOULD send a `READ` command to a client in the following cases:
+- if the `soju.im/read` capability is negotiated, after it sends a server `JOIN` command to the client for a corresponding channel
+- if the `soju.im/read` capability is negotiated, after the last read timestamp of the target changes
+
+If the last read timestamp for a target is unknown or not yet set, it SHOULD not be sent.
+
+#### Read timestamp notes
+
+The last read timestamp of a target SHOULD only ever increase. If a client sends a `READ` command with a timestamp that is below the current known timestamp of the server, the server SHOULD reply with a `READ` command with the newer, previous value that was stored and ignore the client timestamp.
+
+#### Errors and Warnings
+
+Errors are returned using the standard replies syntax.
+
+If the server receives a `READ` command with missing parameters, the `NEED_MORE_PARAMS` error code MUST be returned.
+
+    FAIL READ NEED_MORE_PARAMS :Missing parameters
+
+If the selectors or limit supplied were invalid, the `INVALID_PARAMS` error code SHOULD be returned.
+
+    FAIL READ INVALID_PARAMS [invalid_parameters] :Invalid parameters
+
+If the read timestamp cannot be set or returned due to an error, the `MESSAGE_ERROR` error code SHOULD be returned.
+
+    FAIL READ MESSAGE_ERROR the_given_target [extra_context] :The read timestamp could not be set
+
+### Examples
+
+Updating the read timestamp after the user receives and reads a message
+~~~~
+[s] @2019-01-04T14:33:26.123Z :nick!ident@host PRIVMSG #channel :message
+[c] READ #channel timestamp=2019-01-04T14:33:26.123Z
+[s] :irc.host READ #channel timestamp=2019-01-04T14:33:26.123Z
+~~~~
+
+Getting the read timestamp automatically after joining a channel when the capability is negotiated
+~~~~
+[s] :nick!ident@host JOIN #channel
+[s] :irc.host READ #channel timestamp=2019-01-04T14:33:26.123Z
+~~~~
+
+## Use Cases
+
+Clients can know whether a user has already read newly received messages. For clients that display notifications about new messages or highlights, knowing when messages have been read can enable them to clear notifications for messages that were already read on another device.
+
+Clients never have to actively get the read timestamp because it is provided to them on join and as updated by the server.
```

Comment:
To be merged with `## Description`, or make a `## Motivation` section at the beginning of the spec.